### PR TITLE
Safer user liquidations

### PR DESCRIPTION
**Pull request by mootz12**

### CODE REVIEW

File: test/work_submitter.test.ts [hunk 1]
```diff
@@ -18,141 +19,256 @@ jest.mock('../src/utils/logger.js', () => ({
...
     };
 
     const result = await workSubmitter.submit(submission);
-    const expectedLogMessage = `Successfully submitted bad debt transfer for user: ${submission.user}`;
     expect(result).toBe(true);
     expect(mockedSorobanHelper.submitTransaction).toHaveBeenCalled();
-    expect(logger.info).toHaveBeenCalledWith(expectedLogMessage);
-    expect(sendSlackNotification).toHaveBeenCalledWith(expectedLogMessage);
+    expect(logger.info).toHaveBeenCalled();
+    expect(sendSlackNotification).toHaveBeenCalled();
```
Comment (mootz12) [inline test/work_submitter.test.ts:212]: omitted validation of log messages in favor of just validating that logs are created in general.

File: src/work_submitter.ts [hunk 2]
```diff
@@ -80,10 +76,27 @@ export class WorkSubmitter extends SubmissionQueue<WorkSubmission> {
       await sendSlackNotification(logMessage);
       return true;
     } catch (e: any) {
+      // if pool throws a "LIQ_TOO_SMALL" or "LIQ_TOO_LARGE" error, adjust the fill percentage
+      // by 1 percentage point before retrying.
+      if (e instanceof ContractError) {
+        console.log(e);
```
Comment (Ryang-21) [inline src/work_submitter.ts:82]: Testing fragment

File: test/work_submitter.test.ts [hunk 3]
```diff
@@ -18,141 +19,256 @@ jest.mock('../src/utils/logger.js', () => ({
...
   let mockedSorobanHelperConstructor = SorobanHelper as jest.MockedClass<typeof SorobanHelper>;
-
   const mockedSendSlackNotif = sendSlackNotification as jest.MockedFunction<
     typeof sendSlackNotification
   >;
 
   beforeEach(() => {
-    jest.clearAllMocks();
-    mockDb = inMemoryAuctioneerDb();
-    workSubmitter = new WorkSubmitter(mockDb);
```
Comment (Ryang-21) [inline test/work_submitter.test.ts:36]: workSubmitter needs to be created

**Review by Ryang-21 — COMMENTED** [review]
work_handler.test.ts on ln 99 needs to change from 2 to 3

**Review by Ryang-21 — APPROVED** [review]

**Review by markuspluna — COMMENTED** [review]
I think this looks fine

**Pull request merged**